### PR TITLE
Field polling - new PR

### PR DESCRIPTION
**Pull request by reutshal**
## XSOAR [Partner](https://xsoar.pan.dev/docs/partners/become-a-tech-partner)?
- [ ] The title must be in the following format: **[YOUR_PARTNER_ID] short description**.

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
relates: https://github.com/demisto/content/pull/8991

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Demisto
- [ ] 5.0.0
- [ ] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by reutshal** [comment]
@yaron-libman Can you please go over these files:
Playbook - https://github.com/demisto/content/blob/FieldPolling-newBranch/Packs/CommonPlaybooks/Playbooks/playbook-Field_Polling.yml
Playbook readme - https://github.com/demisto/content/blob/FieldPolling-newBranch/Packs/CommonPlaybooks/Playbooks/playbook-Field_Polling_README.md
Common playbooks RN - https://github.com/demisto/content/blob/FieldPolling-newBranch/Packs/CommonPlaybooks/ReleaseNotes/1_8_1.md
CommonScripts RN - https://github.com/demisto/content/blob/FieldPolling-newBranch/Packs/CommonScripts/ReleaseNotes/1_2_44.md
Script yml - https://github.com/demisto/content/blob/FieldPolling-newBranch/Packs/CommonScripts/Scripts/PollingField/PollingField.yml
Script readme - https://github.com/demisto/content/blob/FieldPolling-
newBranch/Packs/CommonScripts/Scripts/PollingField/README.md
Thanks!

File: Packs/CommonScripts/Scripts/PollingField/PollingField.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    else:
+        if len(field_value) > 0:
+            return True
+    return False
+
+
+def poll_field(args: Dict[str, Any]) -> Tuple[str, dict, dict]:
+
+    field = args.get('field')
+    regex = args.get('regex', None)
```
Comment (yaakovi) [inline Packs/CommonScripts/Scripts/PollingField/PollingField.py:21]:
```suggestion
    regex = args.get('regex')
```

File: Packs/CommonScripts/Scripts/PollingField/PollingField.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    else:
+        custom_fields = incident.get('CustomFields', {})
+        if field in custom_fields:
+            data['exists'] = check_field(custom_fields.get(field), regex)
+
+    context = {
+        'PollingCheckField(val.field == obj.field)': data
+    }
+
+    human_readable = "The field exists" if data['exists'] else "The field does not exist"
```
Comment (yaakovi) [inline Packs/CommonScripts/Scripts/PollingField/PollingField.py:43]:
```suggestion
    human_readable = 'The field exists.' if data['exists'] else 'The field does not exist.'
```

File: Packs/CommonScripts/Scripts/PollingField/PollingField.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
+args:
+- description: The field to check
+  name: field
+  required: true
+- description: The regex pattern to check the field for (optional).
+  name: regex
```
Comment (yaakovi) [inline Packs/CommonScripts/Scripts/PollingField/PollingField.yml:6]: I would add an option for case-insensitive too

File: Packs/CommonScripts/Scripts/PollingField/PollingField_test.py [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+        - The regex sent is matching the field value
+    When
+        - mock the server response to demisto.incidents().
+    Then
+        Validate the script finds the field
+    """
+    mocker.patch.object(demisto, 'incidents', return_value=[incident_contains_field_in_custom_fields])
+    args = {
+        'field': 'field_name',
+        'regex': 'Test'
```
Comment (yaakovi) [inline Packs/CommonScripts/Scripts/PollingField/PollingField_test.py:174]: non of your regexes tests are really a regex...

File: Packs/CommonPlaybooks/Playbooks/playbook-Field_Polling_README.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+| frequency | How often to check \(in minutes\) | 1 | Required |
+| timeout | When to timeout \(in minutes\) | 10 | Required |
+
+## Playbook Outputs
+---
+There are no outputs for this playbook.
+
+## Playbook Image
+---
+![Field Polling Generic](Insert the link to your image here)
```
Comment (yaakovi) [inline Packs/CommonPlaybooks/Playbooks/playbook-Field_Polling_README.md:34]: update link

**Review by yaakovi — COMMENTED** [review]
cool feature

File: Tests/conf.json [hunk 6]
```diff
@@ -3078,7 +3083,6 @@
         "pyEWS_Test": "Issue 28339"
     },
     "skipped_integrations": {
-
```
[the remark this answers is not in the document]
Comment (reutshal) [reply Tests/conf.json:3081]: Reverting, not sure where this came from

File: Packs/CommonPlaybooks/Playbooks/playbook-Field_Polling_README.md (hunk 5 above)
Comment (reutshal) [reply Packs/CommonPlaybooks/Playbooks/playbook-Field_Polling_README.md:34]: Will do

**Comment by reutshal** [comment]
@yaron-libman I also wanted to get your opinion regarding the script's name - I was thinking maybe 'CheckField' or something of that sort would be a clearer name. what do you think?

**Comment by yaron-libman** [comment]
@reutshal re the script name, how about CheckFieldValue? The script checks that the field has a value and what that value is.

**Comment by reutshal** [comment]
> @reutshal re the script name, how about CheckFieldValue? The script checks that the field has a value and what that value is.

Sounds great. I will change it. Thanks!

File: Packs/CommonScripts/Scripts/CheckFieldValue/CheckFieldValue_test.py [hunk 7]
```diff
@@ -0,0 +1,244 @@
...
+        - An incident with the field named 'field_name' with value 'Test' in the 'CustomFields' entry
+        - No regex argument sent to the command
+    When
+        - mock the server response to demisto.incidents().
+    Then
+        Validate the script finds the field
+    """
+    mocker.patch.object(demisto, 'incidents', return_value=[incident_contains_field_in_custom_fields])
+    args = {
+        'field': 'field_name'
```
Comment (yaakovi) [inline Packs/CommonScripts/Scripts/CheckFieldValue/CheckFieldValue_test.py:217]: here and everywhere
```suggestion
        'field': 'field_name',
```

**Review by yaakovi — APPROVED** [review]
🆒

**Pull request merged**